container/layer_tools.bzl

```diff
@@ -255,10 +286,10 @@ def incremental_load(
                 # It is notable that the only legal use of '{' in a
                 # tag would be for stamp variables, '$' is not allowed.
                 tag_reference,
-                _get_runfile_path(ctx, image["config_digest"]),
+                image["config_digest"].path if action_run else _get_runfile_path(ctx, image["config_digest"]),
             ),
         ]
-        if run:
+        if run or commit or extract:
             # Args are embedded into the image, so omitted here.
             run_statements += [
                 "\"${DOCKER}\" ${DOCKER_FLAGS} run %s %s" % (run_flags, tag_reference),
```

Comment:
This causes a small difference in behaviour: before the resulting command was `docker run ... $commands`, now it's `docker run ...` and the commands are taken from the image.